warehouse/forklift/legacy.py

```diff
@@ -1345,6 +1347,21 @@ def file_upload(request):
         file_data = file_
         request.db.add(file_)
 
+        project.record_event(
```

Comment:
Instead of recording these as `ProjectEvents`, I think we'll instead want to extend the `HasEvents` mixin to the `File` model, so this would become `file_.record_event(...)` and we could later do a `joinedload` to call `file_.events` to get all events that belong to each file for a release.

We might also want to find a way to have `<Project>.events` eventually include events of sub-models like releases and files, but for now I think that would be sufficient.

Reply:
SGTM! Any thoughts on how `File.events` should be rendered in the project's security history? I figure I can just return all event streams under the project and interleave them by timestamp, but not sure if you have something else in mind.

Reply:
That could work if we want to keep`<Project>.events` to just `ProjectEvents`. I'm somewhat concerned that this could become an N+1 query, but as long as we can construct a query that does a joinedload for _all_ event types, I think it will be fine.

Reply:
I'm not sure if you can do a joined load like that, but you can do a union query for all event types.